dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpServlet.java

```diff
@@ -204,6 +208,7 @@ private void callToolSynchronously(McpTransport transport,
                     throws IllegalAccessException, IllegalArgumentException {
 
         ToolArguments toolArgs = createToolArguments(params);
+        toolArgs.args().put(EncoderRegistry.class.getName(), encoderRegistry);
```

Comment:
Unfortunately, doing it this way is going to mean that the user sees the `EncoderRegistry` as an argument if they list all the entries in the `args()` map.

However, we can give `ToolArgsImpl` an extra field with the `EncoderRegistry`, and cast the `ToolArgs` to `ToolArgsImpl` in our handler.

The other way to do it would be to pass `EncoderRegistry` in the constructor of the handler, but at the moment I think we create the handler before we are able to create instances of beans so that would be a bit more difficult.